src/ansys/tools/visualization_interface/backends/pyvista/pyvista.py

```diff
@@ -284,6 +310,35 @@ def picker_callback(self, actor: "pv.Actor") -> None:
                 self.unselect_object(edge)
                 actor.prop.color = Color.EDGE.value
 
+    def hover_callback(self, _widget, event_name) -> None:
+        """Define the callback for the element hover.
+
+        Parameters
+        ----------
+        actor : ~pyvista.Actor
+            Actor to hover for the picker.
+
+        """
+        plotter = self._pl.scene
+        x, y = plotter.iren.interactor.GetEventPosition()
+        renderer = plotter.iren.get_poked_renderer(x, y)
+        self._hover_picker.Pick(x, y, 0, renderer)
+        actor = self._hover_picker.GetActor()
+        if event_name == "TimerEvent" and actor is not None and actor in self._object_to_actors_map:
+            custom_object = self._object_to_actors_map[actor]
+            label_actor = self._pl.scene.add_point_labels(
+                [actor.GetCenter()],
+                [custom_object.name],
+                always_visible=True,
+                point_size=0,
+                render_points_as_spheres=False,
+                show_points=False,
+            )
+            self._added_hover_labels.append(label_actor)
+        elif event_name == "EndInteractionEvent":
+            for label in self._added_hover_labels:
+                self._pl.scene.remove_actor(label)
```

Comment:
I did noticed that the hovering wasn't super smooth. Probably because you are plotting the labels in every iteration (assuming ``hover_callback`` is called in every mouse movement).
You might want to use a class flag to tell if they have been plotted or not, and then just replot if the ``actor.GetCenter`` change.